// ng2-search-table/services/search-table.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var http_1 = require("@angular/http");
var SearchTableService = (function () {
    function SearchTableService(http) {
        this.http = http;
    }
    // [Sort Params]
    // {
    //    "something": "desc",
    //    "another": "asc"
    // }
    //
    // [Filter Params]
    // {
    //    "something": "someValue",
    //    "another": "1"
    // }
    SearchTableService.prototype.search = function (searchUrl, options) {
        var page = options && options.page || 1;
        var per = options && options.per || 20;
        var sort = options && options.sort;
        var filter = options && options.filter;
        var searchParams = new http_1.URLSearchParams();
        searchParams.set("page", page.toString());
        searchParams.set("per", per.toString());
        if (sort) {
            Object.keys(sort).forEach(function (key) {
                searchParams.set("sort[" + key + "]", sort[key]);
            });
        }
        if (filter) {
            Object.keys(filter).forEach(function (key) {
                searchParams.set("filter[" + key + "]", filter[key]);
            });
        }
        var requestOptions = new http_1.RequestOptions({
            search: searchParams
        });
        return this
            .http
            .get(searchUrl, requestOptions)
            .map(function (r) { return r.json(); })
            .map(function (r) {
            return {
                page: r.page,
                per: r.per,
                totalCount: r.totalCount,
                results: r.results
            };
        });
    };
    SearchTableService = __decorate([
        core_1.Injectable(), 
        __metadata('design:paramtypes', [http_1.Http])
    ], SearchTableService);
    return SearchTableService;
}());
exports.SearchTableService = SearchTableService;
