// ng2-search-table/components/table-filter/from-to-text-filter.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var FromToTextFilterComponent = (function () {
    function FromToTextFilterComponent() {
        this.model = {};
        this.eventEmitter = new core_1.EventEmitter();
        this.isHidden = false;
    }
    FromToTextFilterComponent.prototype.ngOnInit = function () {
        this.termFromControl = new forms_1.FormControl();
        this.termToControl = new forms_1.FormControl();
        this.setValueChangeEmitter(this.termFromControl, this.model.multipleFilter[0].name);
        this.setValueChangeEmitter(this.termToControl, this.model.multipleFilter[1].name);
    };
    FromToTextFilterComponent.prototype.setValue = function (name, value) {
        if (this.model.multipleFilter[0].name === name) {
            this.fromTerm = value;
        }
        else if (this.model.multipleFilter[1].name === name) {
            this.toTerm = value;
        }
    };
    FromToTextFilterComponent.prototype.setVisibility = function (visible) {
        this.isHidden = !visible;
    };
    FromToTextFilterComponent.prototype.setValueChangeEmitter = function (control, controlName) {
        var _this = this;
        control.valueChanges.subscribe(function (newValue) {
            var value = (control === _this.termFromControl ? _this.fromTerm : _this.toTerm) || "";
            _this.eventEmitter.emit({
                value: value,
                model: _this.model,
                name: controlName
            });
        });
    };
    __decorate([
        core_1.HostBinding("hidden"), 
        __metadata('design:type', Boolean)
    ], FromToTextFilterComponent.prototype, "isHidden", void 0);
    FromToTextFilterComponent = __decorate([
        core_1.Component({
            moduleId: module.id,
            selector: "th[from-to-text-filter]",
            inputs: ["name", "model", "debounceMillSeconds"],
            outputs: ["eventEmitter"],
            template: "\n    <div class=\"row\">\n      <div class=\"col-xs-6\">\n        <input class=\"form-control input-sm\"\n               [formControl]=\"termFromControl\"\n               [(ngModel)]=\"fromTerm\"\n               placeholder=\"{{model.multipleFilter[0].placeholder}}\" />\n      </div>\n      <div class=\"col-xs-6\">\n        <input class=\"form-control input-sm\"\n               [formControl]=\"termToControl\"\n               [(ngModel)]=\"toTerm\"\n               placeholder=\"{{model.multipleFilter[1].placeholder}}\"/>\n      </div>\n    </div>\n  "
        }), 
        __metadata('design:paramtypes', [])
    ], FromToTextFilterComponent);
    return FromToTextFilterComponent;
}());
exports.FromToTextFilterComponent = FromToTextFilterComponent;
