// ng2-search-table/components/table-filter/no-filter.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var NoFilterComponent = (function () {
    function NoFilterComponent() {
        this.model = {};
        this.eventEmitter = new core_1.EventEmitter();
        this.isHidden = false;
    }
    NoFilterComponent.prototype.setValue = function (name, value) {
        // nothing
    };
    NoFilterComponent.prototype.setVisibility = function (visible) {
        this.isHidden = !visible;
    };
    __decorate([
        core_1.HostBinding("hidden"), 
        __metadata('design:type', Boolean)
    ], NoFilterComponent.prototype, "isHidden", void 0);
    NoFilterComponent = __decorate([
        core_1.Component({
            moduleId: module.id,
            selector: "th[no-filter]",
            inputs: ["name", "model"],
            outputs: ["eventEmitter"],
            template: "\n  "
        }), 
        __metadata('design:paramtypes', [])
    ], NoFilterComponent);
    return NoFilterComponent;
}());
exports.NoFilterComponent = NoFilterComponent;
