// ng2-search-table/components/table-filter/select-filter.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var SelectFilterComponent = (function () {
    function SelectFilterComponent() {
        this.model = {};
        this.eventEmitter = new core_1.EventEmitter();
        this.isHidden = false;
    }
    SelectFilterComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.selectControl = new forms_1.FormControl();
        this.selectControl.valueChanges.subscribe(function (v) {
            var newValue = v || "";
            _this.eventEmitter.emit({
                value: newValue,
                model: _this.model,
                name: _this.name
            });
        });
    };
    SelectFilterComponent.prototype.setValue = function (name, value) {
        this.selectValue = value;
    };
    SelectFilterComponent.prototype.setVisibility = function (visible) {
        this.isHidden = !visible;
    };
    __decorate([
        core_1.HostBinding("hidden"), 
        __metadata('design:type', Boolean)
    ], SelectFilterComponent.prototype, "isHidden", void 0);
    SelectFilterComponent = __decorate([
        core_1.Component({
            moduleId: module.id,
            selector: "th[select-filter]",
            inputs: ["name", "model"],
            outputs: ["eventEmitter"],
            template: "\n  <select class=\"form-control input-sm\"\n          [formControl]=\"selectControl\" [(ngModel)]=\"selectValue\">\n    <option *ngFor=\"let cond of model.selectValues\" value=\"{{cond.id}}\">\n      {{cond.name}}\n    </option>\n  </select>\n  "
        }), 
        __metadata('design:paramtypes', [])
    ], SelectFilterComponent);
    return SelectFilterComponent;
}());
exports.SelectFilterComponent = SelectFilterComponent;
