// ng2-search-table/components/table-filter/text-filter.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var TextFilterComponent = (function () {
    function TextFilterComponent() {
        this.model = {};
        this.debounceMillSeconds = 300;
        this.eventEmitter = new core_1.EventEmitter();
        this.isHidden = false;
    }
    TextFilterComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.termControl = new forms_1.FormControl();
        this.termControl.valueChanges.subscribe(function (newValue) {
            var value = newValue || "";
            _this.eventEmitter.emit({
                value: value,
                model: _this.model,
                name: _this.name
            });
        });
    };
    TextFilterComponent.prototype.setValue = function (name, value) {
        this.term = value;
    };
    TextFilterComponent.prototype.setVisibility = function (visible) {
        this.isHidden = !visible;
    };
    __decorate([
        core_1.HostBinding("hidden"), 
        __metadata('design:type', Boolean)
    ], TextFilterComponent.prototype, "isHidden", void 0);
    TextFilterComponent = __decorate([
        core_1.Component({
            moduleId: module.id,
            selector: "th[text-filter]",
            inputs: ["name", "model", "debounceMillSeconds"],
            outputs: ["eventEmitter"],
            template: "\n    <input class=\"form-control input-sm\" [formControl]=\"termControl\" [(ngModel)]=\"term\" />\n  "
        }), 
        __metadata('design:paramtypes', [])
    ], TextFilterComponent);
    return TextFilterComponent;
}());
exports.TextFilterComponent = TextFilterComponent;
