// ng2-search-table/ng2-search-table.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var http_1 = require("@angular/http");
var forms_1 = require("@angular/forms");
var platform_browser_1 = require("@angular/platform-browser");
require("rxjs/Rx");
var search_table_component_1 = require("./components/search-table.component");
var search_table_service_1 = require("./services/search-table.service");
var no_header_component_1 = require("./components/header/no-header.component");
var simple_header_component_1 = require("./components/header/simple-header.component");
var sortable_header_component_1 = require("./components/header/sortable-header.component");
var no_filter_component_1 = require("./components/table-filter/no-filter.component");
var text_filter_component_1 = require("./components/table-filter/text-filter.component");
var select_filter_component_1 = require("./components/table-filter/select-filter.component");
var from_to_text_filter_component_1 = require("./components/table-filter/from-to-text-filter.component");
// exports
var search_table_component_2 = require("./components/search-table.component");
exports.SearchTableComponent = search_table_component_2.SearchTableComponent;
var search_table_service_2 = require("./services/search-table.service");
exports.SearchTableService = search_table_service_2.SearchTableService;
var no_header_component_2 = require("./components/header/no-header.component");
exports.NoHeaderComponent = no_header_component_2.NoHeaderComponent;
var simple_header_component_2 = require("./components/header/simple-header.component");
exports.SimpleHeaderComponent = simple_header_component_2.SimpleHeaderComponent;
var sortable_header_component_2 = require("./components/header/sortable-header.component");
exports.SortableHeaderComponent = sortable_header_component_2.SortableHeaderComponent;
var no_filter_component_2 = require("./components/table-filter/no-filter.component");
exports.NoFilterComponent = no_filter_component_2.NoFilterComponent;
var text_filter_component_2 = require("./components/table-filter/text-filter.component");
exports.TextFilterComponent = text_filter_component_2.TextFilterComponent;
var select_filter_component_2 = require("./components/table-filter/select-filter.component");
exports.SelectFilterComponent = select_filter_component_2.SelectFilterComponent;
var from_to_text_filter_component_2 = require("./components/table-filter/from-to-text-filter.component");
exports.FromToTextFilterComponent = from_to_text_filter_component_2.FromToTextFilterComponent;
var Ng2SearchTableModule = (function () {
    function Ng2SearchTableModule() {
    }
    Ng2SearchTableModule.forRoot = function () {
        return {
            ngModule: Ng2SearchTableModule,
            providers: []
        };
    };
    Ng2SearchTableModule = __decorate([
        core_1.NgModule({
            imports: [
                platform_browser_1.BrowserModule,
                common_1.CommonModule,
                http_1.HttpModule,
                forms_1.FormsModule,
                forms_1.ReactiveFormsModule
            ],
            declarations: [
                search_table_component_1.SearchTableComponent,
                // Header
                no_header_component_1.NoHeaderComponent,
                simple_header_component_1.SimpleHeaderComponent,
                sortable_header_component_1.SortableHeaderComponent,
                // Filter
                no_filter_component_1.NoFilterComponent,
                text_filter_component_1.TextFilterComponent,
                select_filter_component_1.SelectFilterComponent,
                from_to_text_filter_component_1.FromToTextFilterComponent
            ],
            providers: [
                search_table_service_1.SearchTableService
            ],
            exports: [
                search_table_component_1.SearchTableComponent,
                // Header
                no_header_component_1.NoHeaderComponent,
                simple_header_component_1.SimpleHeaderComponent,
                sortable_header_component_1.SortableHeaderComponent,
                // Filter
                no_filter_component_1.NoFilterComponent,
                text_filter_component_1.TextFilterComponent,
                select_filter_component_1.SelectFilterComponent,
                from_to_text_filter_component_1.FromToTextFilterComponent
            ]
        }), 
        __metadata('design:paramtypes', [])
    ], Ng2SearchTableModule);
    return Ng2SearchTableModule;
}());
exports.Ng2SearchTableModule = Ng2SearchTableModule;
