// ng2-search-table/components/header/sortable-header.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var SortableHeaderComponent = (function () {
    function SortableHeaderComponent() {
        this.model = {};
        this.eventEmitter = new core_1.EventEmitter();
        this.isHidden = false;
    }
    Object.defineProperty(SortableHeaderComponent.prototype, "asc", {
        get: function () { return this.model.direction === "asc"; },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SortableHeaderComponent.prototype, "desc", {
        get: function () { return this.model.direction === "desc"; },
        enumerable: true,
        configurable: true
    });
    SortableHeaderComponent.prototype.onToggleSort = function () {
        switch (this.model.direction) {
            case "asc":
                this.model.direction = "desc";
                break;
            case "desc":
                this.model.direction = "asc";
                break;
            default:
                this.model.direction = "asc";
                break;
        }
        this.eventEmitter.emit({
            value: this.model.direction,
            model: this.model,
            name: this.name
        });
    };
    SortableHeaderComponent.prototype.setVisibility = function (visible) {
        this.isHidden = !visible;
    };
    __decorate([
        core_1.HostBinding("hidden"), 
        __metadata('design:type', Boolean)
    ], SortableHeaderComponent.prototype, "isHidden", void 0);
    __decorate([
        core_1.HostBinding("class.sort-asc"), 
        __metadata('design:type', Object)
    ], SortableHeaderComponent.prototype, "asc", null);
    __decorate([
        core_1.HostBinding("class.sort-desc"), 
        __metadata('design:type', Object)
    ], SortableHeaderComponent.prototype, "desc", null);
    SortableHeaderComponent = __decorate([
        core_1.Component({
            moduleId: module.id,
            selector: "th[simple-sortable-header]",
            inputs: ["name", "model"],
            outputs: ["eventEmitter"],
            host: {
                "(click)": "onToggleSort()"
            },
            template: "\n    {{model.displayName}}\n    <i class=\"pull-right fa\"\n    [ngClass]=\"{'fa-sort': (model.direction != 'asc' && model.direction != 'desc'),\n      'fa-sort-asc': model.direction == 'asc',\n      'fa-sort-desc': model.direction == 'desc'}\"\n    aria-hidden='true'></i>\n  "
        }), 
        __metadata('design:paramtypes', [])
    ], SortableHeaderComponent);
    return SortableHeaderComponent;
}());
exports.SortableHeaderComponent = SortableHeaderComponent;
